### PR TITLE
Issue 17075 personalization permissions

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java [hunk 1]
```diff
@@ -89,6 +98,19 @@ public Response personalizePageContainers (@Context final HttpServletRequest  re
...
+        final String pageId = personalizationPersonaPageForm.getPageId();
+
+        if (!UtilMethods.isSet(pageId)) {
+            throw new BadRequestException(
+                    "Page parameter is missing");
+        }
+
+        final Contentlet pageContentlet = contentletAPI.findContentletByIdentifierAnyLanguage(pageId);
+        if(!permissionAPI.doesUserHavePermission(pageContentlet, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndRoles)){
+            Logger.error(PersonalizationResource.class,String.format("User `%s` does not have edit permission over page `%s` therefore personalization isn't allowed.  ",user.getUserId(), pageId));
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java]: This looks great - only one thing, lets change the `Logger.error` to `Logger.warns`

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java]: done!

**Review by wezell — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java [hunk 2]
```diff
@@ -89,6 +98,19 @@ public Response personalizePageContainers (@Context final HttpServletRequest  re
...
+
+        if (!UtilMethods.isSet(pageId)) {
+            throw new BadRequestException(
+                    "Page parameter is missing");
+        }
+
+        final Contentlet pageContentlet = contentletAPI.findContentletByIdentifierAnyLanguage(pageId);
+        if(!permissionAPI.doesUserHavePermission(pageContentlet, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndRoles)){
+            Logger.warn(PersonalizationResource.class,String.format("User `%s` does not have edit permission over page `%s` therefore personalization isn't allowed.  ",user.getUserId(), pageId));
+            return Response.status(Status.UNAUTHORIZED).build();
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java]: When no permissions we use `Status.FORBIDDEN`

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java [hunk 3]
```diff
@@ -132,6 +154,12 @@ public Response personalizePageContainers (@Context final HttpServletRequest  re
             throw new BadRequestException("Does not exists a Persona with the tag: " + personalization);
         }
 
+        final Contentlet pageContentlet = contentletAPI.findContentletByIdentifierAnyLanguage(pageId);
+        if(!permissionAPI.doesUserHavePermission(pageContentlet, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndRoles)){
+            Logger.warn(PersonalizationResource.class,String.format("User `%s` does not have edit permission over page `%s` therefore personalization isn't allowed.  ",user.getUserId(), pageId));
+            return Response.status(Status.UNAUTHORIZED).build();
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java]: Same as above

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java]: done!

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/personalization/PersonalizationResource.java]: Same as above

**Review by jgambarios — APPROVED** [review]

**Pull request merged**